scripts/dumpapp

```diff
@@ -24,28 +22,17 @@ def main():
   # transport.
   device = os.environ.get('ANDROID_SERIAL')
   try:
-    conn = HTTPConnectionOverADB(device, process)
-    query_params = ['argv=' + urllib.parse.quote(arg) for arg in args]
-    url = '/dumpapp?%s' % ('&'.join(query_params))
-    fake_host = { 'Host': 'localhost:5037' }
+    sock = stetho_open(device, process)
 
-    http_method = None
-    body = None
-    if not sys.stdin.isatty():
-      http_method = 'POST'
-      body = sys.stdin.detach().read()
-    else:
-      http_method = 'GET'
-      body = None
-
-    conn.request(http_method, url, body, fake_host)
-
-    reply = conn.getresponse()
+    enter_frame = b'!' + struct.pack('!L', len(args))
+    for arg in args:
+      enter_frame += struct.pack(
+          '!H' + str(len(arg)) + 's',
+          len(arg),
+          arg.encode('utf-8'))
```

Comment:
We should have a non-trivial utf-8 test if we don't already.  It seems surprising here that we would send the length of the pre-encoded string rather than the number of utf-8 bytes.